server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -219,15 +226,37 @@ public void restoreSnapshot(final RestoreRequest request, final ActionListener<R
                 validateSnapshotRestorable(repositoryName, snapshotInfo);
 
                 // Resolve the indices from the snapshot that need to be restored
-                final List<String> indicesInSnapshot = request.includeIndices()
-                    ? filterIndices(snapshotInfo.indices(), request.indices(), request.indicesOptions())
+
+                List<String> resolvedIndices = new ArrayList<>();
+                List<String> resolvedTemplates = new ArrayList<>();
+
+                resolveIndices(
+                    request,
+                    tablesToRestore,
+                    snapshotInfo.indices(),
+                    clusterService.state().nodes().getMinNodeVersion(),
+                    resolvedIndices,
+                    resolvedTemplates
+                );
+
+                boolean includeIndices = request.includeIndices();
+                if (includeIndices) {
+                    // See https://github.com/crate/crate/issues/14144
+                    if (resolvedIndices.isEmpty()) {
+                        includeIndices = false;
+                    }
+                }
```

Comment:
This is because in ES empty list is treated as "select all available"

See `IndexNameExpressionResolver.isAllIndices(selectedIndices)` 
and also JavaDoc for ` public RestoreSnapshotRequest indices(String... indices) {`


Keeping both this fix + empty list == all looks a bit contradicting. Would it be fine to stop treating an empty list as "all"? It looks more logical : nothing selected == nothing resolved.

Reply:
Actually, this is wrong adaption of 
https://github.com/crate/crate/pull/14163/commits/f90e3927db030b7b260ce481b125708b14de9d91,

 after this change `test_restore_empty_partitioned_table_does_not_restore_other_tables` passes but many other tests fail.

Reply:
Pushed a fixup https://github.com/crate/crate/pull/15071/commits/f3175d4ccc1f7d00d398718a9a3ad4b043c184d0